src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileItem.cs

```diff
@@ -18,7 +18,19 @@ public LockFileItem(string path)
 
         public string Path { get; }
 
-        public IDictionary<string, string> Properties { get; } = new Dictionary<string, string>();
+        private Dictionary<string, string> _properties;
+        public IDictionary<string, string> Properties
+        {
+            get
+            {
+                if (_properties == null)
+                {
+                    _properties = new Dictionary<string, string>();
```

Comment:
DIdn't know what the threading semantics are around this. Should I go ahead and put in a lock such that multiple calls into the same object can't possible both create a dictionary?

Reply:
In practice, the construction/reading are single threaded. 
The safer approach might be a lock though.

Reply:
Went ahead and put this behind a static lock